packages/cw721/src/execute.rs

```diff
@@ -0,0 +1,886 @@
+use cosmwasm_std::{
+    Addr, Api, BankMsg, Binary, Coin, CustomMsg, Deps, DepsMut, Empty, Env, MessageInfo, Response,
+    StdResult, Storage,
+};
+use cw_ownable::{none_or, Action, Ownership, OwnershipError};
+use cw_storage_plus::Item;
+use cw_utils::Expiration;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+use crate::{
+    error::Cw721ContractError,
+    msg::{
+        CollectionInfoExtensionMsg, CollectionInfoMsg, Cw721ExecuteMsg, Cw721InstantiateMsg,
+        Cw721MigrateMsg,
+    },
+    receiver::Cw721ReceiveMsg,
+    state::{
+        CollectionInfo, Cw721Config, DefaultOptionCollectionInfoExtension,
+        DefaultOptionMetadataExtension, EmptyMsg, NftInfo, CREATOR, MINTER,
+    },
+    Approval, RoyaltyInfo,
+};
+
+pub trait Validate {
+    fn validate(&self) -> Result<(), Cw721ContractError>;
+}
+
+impl Validate for Empty {
+    fn validate(&self) -> Result<(), Cw721ContractError> {
+        Ok(())
+    }
+}
+
+pub trait Update<T>: Sized {
+    fn update(&self, msg: &T) -> Result<Self, Cw721ContractError>;
+}
+
+impl Update<EmptyMsg> for Empty {
+    fn update(&self, _msg: &EmptyMsg) -> Result<Self, crate::error::Cw721ContractError> {
+        Ok(Empty::default())
+    }
+}
+
+impl Update<EmptyMsg> for Option<Empty> {
+    fn update(&self, _msg: &EmptyMsg) -> Result<Self, crate::error::Cw721ContractError> {
+        match self {
+            Some(ext) => Ok(Some(ext.clone())),
+            None => Ok(Some(Empty::default())),
+        }
+    }
+}
```

Comment:
2 new useful traits: `Update` and `Validate`. Avoid duplicates like during instantiation and execution.